### DISCUSSION TITLE
How about support lambda in JDK 8 ?

### DISCUSSION THREAD

**Discussion by sunxien** [qos-ch/slf4j#352] [Ideas]
See `Logger.class` provide APIs like below:

```
void info(Supplier<?> messageSupplier);

void info(String message, Supplier<?>... paramSuppliers);
.... 

and other APIs in different log level
```

_**Best Regards** from Eric Sun_

**Comment by sunxien** [comment]
Developer will use

`log.debug(“Now time is {}”, () -> new SimpleDateFormat(“yyyy-MM-dd”).format(new Date()));`

instead of

`if (log.isDebugEnable()) {`
`    log.debug(“Now time is {}”, new SimpleDateFormat(“yyyy-MM-dd”).format(new Date()));`
`}`

As 

` log.debug(“Now time is {}”, new SimpleDateFormat(“yyyy-MM-dd”).format(new Date()));`

may cause performance issue when format date is running.

**Comment by ceki** [comment]
@sunxien Have you seen the [fluent API in SLF4J 2.0]( https://www.slf4j.org/manual.html#fluent) which addresses this issue?

**Reply by sunxien** [reply]
Yep, the fluent API have the same effect as lambda. Thanks CEKI.
